Userland/Libraries/LibVideo/VP9/Decoder.cpp

```diff
@@ -908,9 +892,7 @@ DecoderErrorOr<void> Decoder::predict_inter_block(u8 plane, u8 ref_list, u32 x,
     // The filtering is applied as follows:
     // The array intermediate is specified as follows:
     // Note: Height is specified by `intermediate_height`, width is specified by `width`
-    Vector<u16>& intermediate_buffer = m_buffers.inter_horizontal;
-    intermediate_buffer.clear_with_capacity();
-    intermediate_buffer.resize_and_keep_capacity(intermediate_height * width);
+    u16 intermediate_buffer[intermediate_height * width];
```

Comment:
FixedArray, or we could actually use an arena allocator...

Reply:
or another `AK::Array` with a upper bound

Reply:
Upper bound works here too, I must've missed this on my last search through for VLAs.